src/item.cpp

```diff
@@ -3998,20 +3992,33 @@ int item::gun_pierce( bool with_ammo ) const
     return ret;
 }
 
-int item::gun_recoil( bool with_ammo ) const
+int item::gun_handling() const
 {
     if( !is_gun() ) {
         return 0;
+
     }
-    int ret = type->gun->recoil;
-    if( with_ammo && ammo_data() ) {
-        ret += ammo_data()->ammo->recoil;
-    }
+    double handling = type->gun->handling;
     for( const auto mod : gunmods() ) {
-        ret += mod->type->gunmod->recoil;
+        handling += mod->type->gunmod->handling;
     }
-    ret += damage() * 15;
-    return ret;
+
+    return std::max( pow( ( weight() / 250 ), 0.8 ) * pow( handling / 10.0, 1.2 ), 0.0 );
```

Comment:
Does this formula have to be this complex?
Can't the weight part be extracted to a json value calculated at item load? That would help with inherently accurate guns, without having to give them giant handling, which in turn wouldn't make their resulting handling get huge if you slapped a lead block onto the gun.

Would be better to go with linear where possible. Low powers (near 1) are confusing.
The extra accuracy gained here may cause worse feedback due to players not understanding what is going on.